### PR TITLE
Fix compilation and assert failure

### PR DESCRIPTION
**Pull request by amitanandaiyer**

### CODE REVIEW

**Review by abhinab-yb — APPROVED** [review]

File: src/yb/util/wait_state.h [hunk 1]
```diff
@@ -12,6 +12,7 @@
 //
 #pragma once
 
+#include <array>
```
Comment (hbhanawat) [inline src/yb/util/wait_state.h:15]: @amitanandaiyer is this change needed?

Comment (amitanandaiyer) [reply src/yb/util/wait_state.h:15]: Yes. Abhinab has used an array in this file.

The compiler on my laptop complains/requires this. Not sure why others don't complain.
